### ISSUE TITLE
phx-click-away does not respect phx-target

### ISSUE THREAD

**Issue by whitepaperclip** [phoenixframework/phoenix_live_view#1808]
### Environment

* Elixir version (elixir -v): 1.12.3
* Phoenix version (mix deps): 1.6.5
* Phoenix LiveView version (mix deps): 0.17.5
* Operating system: macOS
* Browsers you attempted to reproduce this bug on (the more the merrier): Chrome
* Does the problem persist after removing "assets/node_modules" and trying again? Yes/no: yes

### Actual behavior

phx-click-away does not respect phx-target

For example, writing the following in a LiveComponent:

`<div phx-click-away="event" phx-target={@myself}>`

causes the event to be sent to the parent LiveView. `phx-click`, on the other hand, does send it to the component.

### Expected behavior

I expect the event to be sent to the LiveComponent.

**Comment by maxvw** [comment]
I looked into this today and it looks like the code responsible for this has been changed since the last release (as of writing, v0.17.5).

Before:
https://github.com/phoenixframework/phoenix_live_view/blob/e5084d93221c1f11114f95d83d675b5ce4a51578/assets/js/phoenix_live_view/live_socket.js#L576-L588

After:
https://github.com/phoenixframework/phoenix_live_view/blob/2b118bb22fa3b489643772609ca254fe22787d9e/assets/js/phoenix_live_view/live_socket.js#L587-L602

In 0.17.5 it looks like it is sending to the `e.target` element, which is where you click and not the element responsible for. Changing this to the `el` variable fixes it.

When trying out master however I encounter a new bug, there is now a private `click-ref` property being set on clicks and if you have multiple elements on the page at the same time that have `click-away` this will block regular `clicks` from working. 

I removed the other element that was breaking it for me to test if the targeting has been fixed in `master`, but no. It tries to get the nearest `phx-click` target to set the private `click-ref` property and is now passing along _that_ target to the `JS.exec` function. But it should probably be `el`, again. At least, that fixes the issue.

However, I'm not sure if `this.eventMeta` should also be using `el` or if `e.target` is fine for that use case.
